js/src/utils/vercel.ts

```diff
@@ -0,0 +1,90 @@
+import { KVMap } from "../schemas.js";
+
+function extractInputTokenDetails(providerMetadata: Record<string, unknown>) {
+  const inputTokenDetails: Record<string, number> = {};
+  if (
+    providerMetadata.anthropic != null &&
+    typeof providerMetadata.anthropic === "object"
+  ) {
+    const anthropic = providerMetadata.anthropic as Record<string, unknown>;
+    if (
+      anthropic.cacheReadInputTokens != null &&
+      typeof anthropic.cacheReadInputTokens === "number"
+    ) {
+      inputTokenDetails.cache_read = anthropic.cacheReadInputTokens;
+    }
+    if (
+      anthropic.cacheCreationInputTokens != null &&
+      typeof anthropic.cacheCreationInputTokens === "number"
+    ) {
+      inputTokenDetails.ephemeral_5m_input_tokens =
+        anthropic.cacheCreationInputTokens;
+    }
+    return inputTokenDetails;
+  } else if (
+    providerMetadata.openai != null &&
+    typeof providerMetadata.openai === "object"
+  ) {
+    const openai = providerMetadata.openai as Record<string, unknown>;
+    if (
+      openai.cachedPromptTokens != null &&
+      typeof openai.cachedPromptTokens === "number"
+    ) {
+      inputTokenDetails.cache_read = openai.cachedPromptTokens;
+    }
+  }
+  return inputTokenDetails;
+}
+
+export function extractUsageMetadata(span: {
+  status?: { code: number };
+  attributes?: Record<string, unknown>;
+}) {
+  const isError = span.status?.code === 2;
```

Comment:
Since we're doing the `foo as KVMap` when calling do we need to do `span?.status` ?

Reply:
Could probably type it strongly but want to be defensive here in case `status` is missing for some reason on a span

Reply:
I mean the opposite - are we defensive enough?

Reply:
I think so but can make the param optional